### PR TITLE
Adjust read me

### PR DESCRIPTION
**Pull request by RogerKSI**
## Implementation details
- adjust readme

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,50 @@
...
+
+```sh
+cd libgo_owasm && cargo build --release
 ```
 
-## Build shared object library
+### Compile shared and static libraries
 
-Currently, we support only build on linux. Osx build will be available.
+Currently, you can build libraries on Linux and OS X with x86_64 architecture only. However, if you are working on a platform that is not currently supported, you can push code into GitHub. Then, GitHub Action will help to test, compile the library, and push it to your branch.
```
Comment (colmazia) [inline README.md:37]: will help testing and compiling the library, then push it to your branch

File: README.md [hunk 2]
```diff
@@ -1,13 +1,50 @@
...
 
-- run `make docker-images` to pre-built dependencies
-- run `make release` to build current code to `libgo_owasm.so` and `libgo_owasm.dylib`
+You can use the below commands to generate libraries for Linux (x86_64) and OS X (x86_64 and aarch64).
```
Comment (colmazia) [inline README.md:39]: commands below

**Comment by colmazia** [comment]
Grammar seems off, summon @warittornc

File: README.md (hunk 1 above)
Comment (RogerKSI) [reply README.md]: updated

File: README.md (hunk 2 above)
Comment (RogerKSI) [reply README.md]: updated

File: README.md [hunk 3]
```diff
@@ -1,13 +1,50 @@
...
+
+The `./build` directory contains scripts and docker files for building the library from Rust code.
+
+### .github
+
+The `./github` directory, section contains GitHub Action code that aids in testing, building libraries, and releasing packages.
+
+## Development
+
+If you make updates to owasm-vm or the Rust code, it's imperative to [generate bindings.sh](#generate-bindingsh-for-go) and [compile libraries](#compile-shared-and-static-libraries) to update the compiled library within the project, thereby ensuring the go-owasm package incorporates the latest version of the Owasm-vm library.
```
Comment (warittornc) [inline README.md:27]:
```suggestion
If you make updates to owasm-vm or the Rust code, it's essential to [generate bindings.sh](#generate-bindingsh-for-go) and [compile libraries](#compile-shared-and-static-libraries) to update the compiled library within the project, thereby ensuring the go-owasm package incorporates the latest version of the Owasm-vm library.
```

File: README.md [hunk 4]
```diff
@@ -1,13 +1,50 @@
...
+
+```sh
+cd libgo_owasm && cargo build --release
 ```
 
-## Build shared object library
+### Compile shared and static libraries
 
-Currently, we support only build on linux. Osx build will be available.
+Currently, libraries can be built on Linux and OS X with x86_64 architecture only. However, if you're operating on an unsupported platform, you can push the code to GitHub. Subsequently, GitHub Actions will facilitate testing, compiling the library, and pushing it to your branch.
```
Comment (warittornc) [inline README.md:37]:
```suggestion
Currently, the libraries can only be built on Linux and OS X using x86_64 architecture. If you are operating on an unsupported platform, you can push the code to GitHub to trigger a suite of GitHub Actions which will test and  compile the library for you.
```

File: README.md [hunk 5]
```diff
@@ -1,13 +1,50 @@
...
 
-- run `make docker-images` to pre-built dependencies
-- run `make release` to build current code to `libgo_owasm.so` and `libgo_owasm.dylib`
+You can use the commands below to generate libraries for Linux (x86_64) and OS X (x86_64 and aarch64).
```
Comment (warittornc) [inline README.md:39]:
```suggestion
You can use the commands below to build the libraries on Linux (x86_64) and OS X (x86_64 and aarch64).
```

Comment (warittornc) [reply README.md:39]: please fix inconsistency on aarch64 here and in the statement prior.

File: README.md [hunk 6]
```diff
@@ -1,13 +1,50 @@
...
+
+```sh
+# Run test
+cd libgo_owasm && cargo test
+
+# Build docker images that are used to compile the Rust code
```
Comment (warittornc) [inline README.md:45]:
```suggestion
# Build docker images used to compile the Rust code
```

File: README.md [hunk 7]
```diff
@@ -1,13 +1,50 @@
...
+make docker-images
+
+# Run those docker images to build libraries
```
Comment (warittornc) [inline README.md:48]:
```suggestion
# Run the docker images to build the libraries
```

**Review by colmazia — APPROVED** [review]
approved readme, but there is a test failing

**Review by warittornc — APPROVED** [review]
approved, but failed test needs to be fixed in another PR

**Pull request merged**